press/press/doctype/self_hosted_server/self_hosted_server.py

```diff
@@ -395,36 +426,44 @@ def _restore_files(self):
 			log_error("Self Hosted Restore error", server=self.name)
 		self.save()
 
+	def get_hostname(self, server_type):
+		return f"{get_symbolic_name(server_type)}-{self.cluster}-{self.hostname}".lower()
+
 	@frappe.whitelist()
 	def create_proxy_server(self):
 		"""
 		Add a new record to the Proxy Server doctype
 		"""
 		try:
-			server = frappe.new_doc("Proxy Server")
-			server.hostname = self.hostname
-			server.title = self.title
-			server.is_self_hosted = True
-			server.domain = self.domain
-			server.self_hosted_server_domain = self.domain
-			server.team = self.team
-			server.ip = self.ip
-			server.private_ip = self.private_ip
-			server.ssh_user = self.ssh_user
-			server.is_primary = True
-			server.cluster = self.cluster
-			server.ssh_port = self.ssh_port
-			new_server = server.insert()
-			self.agent_password = new_server.get_password("agent_password")
-			self.proxy_server = new_server.name
-			self.proxy_server_ip = self.private_ip
+			proxy_server = frappe.get_doc(
```

Comment:
```suggestion
			proxy_server = frappe.new_doc("Proxy Server",
```

new_doc should be used for new_docs as it respects defaults etc. Also semantically correct.